Subject: [PATCH] Dialer: InCallUI: Enable Sustained Performance Mode

We should make sure that thermals are stable during
active calls, so we should enable Sustained Performance
Mode for the InCallActivity.

As a side effect this will also save power during active
calls.

diff --git a/InCallUI/src/com/android/incallui/InCallActivity.java b/InCallUI/src/com/android/incallui/InCallActivity.java
index eca79f8..36efcb1 100644
--- a/InCallUI/src/com/android/incallui/InCallActivity.java
+++ b/InCallUI/src/com/android/incallui/InCallActivity.java
@@ -35,6 +35,7 @@
 import android.graphics.Point;
 import android.hardware.SensorManager;
 import android.os.Bundle;
+import android.os.PowerManager;
 import android.os.Trace;
 import android.support.design.widget.TabLayout;
 import android.telecom.DisconnectCause;
@@ -192,6 +193,12 @@
 
         getWindow().addFlags(flags);
 
+        PowerManager powerManager = (PowerManager) this.getSystemService(Context.POWER_SERVICE);
+        if (powerManager.isSustainedPerformanceModeSupported()) {
+            // Enable Sustained Performance Mode
+            getWindow().setSustainedPerformanceMode(true);
+        }
+
         // Setup action bar for the conference call manager.
         requestWindowFeature(Window.FEATURE_ACTION_BAR_OVERLAY);
         if (getActionBar() != null) {
